docs/guides/headless-vuejs-tutorial.md

```diff
@@ -0,0 +1,280 @@
+import VueApollo from "vue-apollo"
+
+import "./assets/styles.css";
+import App from './App.vue'
+import apolloClient from './apolloClient'
+
+Vue.config.productionTip = false
+
+Vue.use(VueApollo);
+
+const apolloProvider = new VueApollo({
+  defaultClient: apolloClient
+});
+
+new Vue({
+  apolloProvider,
+  render: h => h(App),
+}).$mount('#app')
+
+
+// import the `apolloClient` that we just created
+import apolloClient from './apolloClient'
+
+Vue.config.productionTip = false
+
+// add `VueApollo`
+Vue.use(VueApollo);
+
+// create a provider with the `apolloClient`
+const apolloProvider = new VueApollo({
+  defaultClient: apolloClient
+});
+
+new Vue({
+  apolloProvider,
+  render: h => h(App),
+}).$mount('#app')
+
+```
+
+Finally let's render some mark up 🙂
+
+Head over to [`App.vue`](https://github.com/webiny/webiny-examples/blob/master/headlesscms-vuejs/src/App.vue) file:
+
+```ts
+<template>
+  <div id="app" class="min-h-screen bg-gray-900 text-gray-100">
+  <Nav />
+   <div class="max-w-3xl mx-auto px-4">
+      <!-- some markup code removed for brevity  -->
+
+      <!-- Article list -->
+      <div class="flex mt-16">
+        <!-- Show loading message while query is loading -->
+        <p v-if="$apollo.queries.listFacts.loading" class="text-green-600">Loading facts ...</p>
+
+       <!-- Show error message if there is an error -->
+        <p v-if="$apollo.queries.listFacts.error" class="text-red-600">Oops! Something went wrong!!</p>
+
+        <div class="flex flex-col">
+          <!-- Loop over facts returned by query -->
+          <div class="mb-16" v-for="fact in listFacts && listFacts.data" v-bind:key="fact.id">
+               <h3 class="mb-4 text-lg text-gray-200 uppercase">
+               {{ fact.title }}
+               </h3>
+               <img :src="fact.image" alt="" class="mb-4"/>
+               <p class="mb-4 text-lg text-gray-300">
+               {{ fact.description }}
+               </p>
+            </div>
+        </div>
+      </div>
+    </div>
+
+  </div>
+</template>
+
+<script>
+
+import gql from "graphql-tag";
+import Nav from "./components/Nav";
+
+export default {
+  name: 'App',
+  components: {
+    Nav
+  },
+  data() {
+    // save `listFacts` returned from GraphQL query
+    return {
+      listFacts: []
+    };
+  },
+  apollo: {
+    // The actual GraphQL query
+    listFacts: gql`
+      query listFacts {
+        listFacts {
+          data {
+            id
+            title
+            description
+            image
+          }
+        }
+      }
+    `
+  }
+}
+</script>
+
+```
+
+## Previewing the page
+
+Let's run `yarn serve` in our Vue.js project directory so we can see our page in action:
+
+![Vue.js Blog](/img/guides/headless-vuejs-tutorial/output.png)
+
+## Conclusion
+
+Congratulations! 🎉
+
+We've have successfully created a simple page that displays a list of all created facts, all powered by Webiny Headless CMS and Vue.js The Progressive JavaScript Framework
```

Comment:
Grammar fixes:
We've successfully created a simple page that displays a list of all created facts, powered by Webiny Headless CMS and Vue.js.